app/src/components/inputs/float_label_input/FloatingLabelInputDemo.tsx

```diff
@@ -0,0 +1,244 @@
+          <Grid>
+            <FloatingLabelInput<DemoForm>
+              id="basicInput"
+              label="Basic Input"
+              register={register}
+            />
+
+            <FloatingLabelInput<DemoForm>
+              id="requiredInput"
+              label="Required Input"
+              register={register}
+              error={errors.requiredInput?.message}
+            />
+
+            <FloatingLabelInput<DemoForm>
+              id="disabledInput"
+              label="Disabled Input"
+              register={register}
+              value="Disabled Input"
+              disabled
+            />
+          </Grid>
+        </Section>
+
+        <Section>
+          <SectionTitle>Validation Examples</SectionTitle>
+          <Grid>
+            <FloatingLabelInput<DemoForm>
+              id="email"
+              label="Email"
+              type="email"
+              register={register}
+              error={errors.email?.message}
+              validation={(value) => {
+                try {
+                  validationSchema.validateSyncAt("email", {
+                    email: value,
+                  });
+                  return true;
+                } catch {
+                  return false;
+                }
+              }}
+            />
+
+            <FloatingLabelInput<DemoForm>
+              id="password"
+              label="Password"
+              type="password"
+              register={register}
+              error={errors.password?.message}
+              validation={(value) => {
+                try {
+                  validationSchema.validateSyncAt("password", {
+                    password: value,
+                  });
+                  return true;
+                } catch {
+                  return false;
+                }
+              }}
+            />
+
+            <FloatingLabelInput<DemoForm>
+              id="maxLengthInput"
+              label="Max Length (20 chars)"
+              register={register}
+              error={errors.maxLengthInput?.message}
+              maxLength={20}
+            />
+          </Grid>
+        </Section>
+
+        <Section>
+          <SectionTitle>Input Types</SectionTitle>
+          <Grid>
+            <FloatingLabelInput<DemoForm>
+              id="numberInput"
+              label="Number Input"
+              type="number"
+              register={register}
+              error={errors.numberInput?.message}
+            />
+
+            <FloatingLabelInput<DemoForm>
+              id="urlInput"
+              label="URL Input"
+              type="url"
+              register={register}
+              error={errors.urlInput?.message}
+            />
+
+            <FloatingLabelInput<DemoForm>
+              id="telInput"
+              label="Phone Number"
+              type="tel"
+              register={register}
+              error={errors.telInput?.message}
+            />
```

Comment:
I understand it's just a demo, however i think we should establish the way of putting static data into data folder and using js map to add repeated html tags with different props, what do you think?

Reply:
Understandable in general but I think it will just add complexity to creating demos like these. after awhile no one will make a demo for his components.